jenkins/Jenkinsfile

```diff
@@ -0,0 +1,246 @@
+def configuration = [vaultUrl: "${VAULT_URL}",  vaultCredentialId: "vault-app-role", engineVersion: 2]
+
+def secrets = [
+  [path: 'secret/jenkins/cloudify', engineVersion: 2, secretValues: [
+    [envVar: 'OS_USERNAME', vaultKey: 'username'],
+    [envVar: 'OS_TENANT', vaultKey: 'tenant'],
+    [envVar: 'OS_IP', vaultKey: 'ip'],
+    [envVar: 'OS_PASSWORD', vaultKey: 'password']]],
+]
+
+@Library('pipeline-shared-library') _
+
+pipeline {
+  agent {
+    kubernetes {
+      label "agent-${env.BUILD_NUMBER}"
+      defaultContainer 'jnlp'
+      yamlFile 'jenkins/build-pod.yaml'
+    }
+  }
+  options {
+    checkoutToSubdirectory('cloudify-agent')
+    buildDiscarder(logRotator(numToKeepStr:'10'))
+    timeout(time: 60, unit: 'MINUTES')
+    timestamps()
+  }
+  environment {
+    VERSION = getVersion("master").trim()
+    PRERELEASE = getPreRelease("master").trim()
+    PROJECT = "cloudify-agent"
+    PATH = "/root/.local/bin:$PATH"
+  }
+  stages{
+    stage ('compatability and flake8') {
+      parallel{
+        stage ('py3_compat'){
+          steps{
+            sh script: "mkdir -p ${env.WORKSPACE}/py3_compat && cp -rf ${env.WORKSPACE}/${env.PROJECT}/. ${env.WORKSPACE}/py3_compat", label: "copying repo to seperate workspace"
+            container('py27'){
+              dir("${env.WORKSPACE}/py3_compat"){
+                echo 'install futurize'
+                sh 'pip install future --user'
+                echo 'find python3-incompatible code'
+                sh '''#!/bin/bash
+                FUTURIZE="futurize ."
```

Comment:
I see we using this 'futurize' thing a lot, maybe we can extract it to a shared library? what you think?

Reply:
That sounds like a great idea, it does come up a lot

Reply:
Should I open a Jira for it?

Reply:
no, extract to a shared library, if you will open the JIRA it will not happen and we using this a lot I see